### PR TITLE
Feature configuration of all referencing speeds

### PR DESCRIPTION
**Pull request by PhilKTurner**
also: configuration validation

### CODE REVIEW

**Comment by PhilKTurner** [comment]
Code style review, @mrinkartiso? :P

**Comment by PhilKTurner** [comment]
Alternatively, @Lukas-Eisele could take a look at this.

File: TxtControllerLibTests/ConfigurationValidatorTests.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        {
+            var configuration = new MotorConfiguration();
+
+            var action = new Action(() => ConfigurationValidator.ValidateMotorConfiguration(configuration));
+
+            action.ShouldThrow<ConfigurationValidationException>("configuration is not initialized");
+        }
+
+        [TestMethod]
+        public void InvalidMotorConfigurations()
```
Comment (Lukas-Eisele) [inline TxtControllerLibTests/ConfigurationValidatorTests.cs]: Too much tested in a single test

**Comment by tziemek** [comment]
Why are you using NotNullAttribute from Resharper annotations? The implementation of the attribute does nothing, hence you have to do explicit checking. The annotation is only useful if you use R#.

**Comment by PhilKTurner** [comment]
@tziemek It is only useful in combination with R#, but it doesn't hurt without R#, does it?

**Comment by tziemek** [comment]
@PhilKTurner Doesn't hurt, but i find it confusing. It suggests static code analysis, but only with R# :-( Don't have it installed atm.

**Pull request merged**